### PR TITLE
Fix handling when `describe_tasks` does not return any tasks during `ECSTask` watch

### PR DESCRIPTION
**Pull request by madkinsz**
Closes https://github.com/PrefectHQ/prefect-aws/issues/151

This is best viewed [without whitespace changes](https://github.com/PrefectHQ/prefect-aws/pull/153/files?diff=unified&w=1)

### Checklist

- [x] This pull request references any related issue by including "Closes #<ISSUE_NUMBER>"
	- If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect-aws/issues/new/choose) first.
- [ ] This pull request includes tests or only affects documentation.
- [x] Summarized PR's changes in [CHANGELOG.md](https://github.com/PrefectHQ/prefect-aws/blob/main/CHANGELOG.md)

### CODE REVIEW

**Comment by madkinsz** [comment]
This one is difficult to cover in tests. We could attempt to mock the first-return of `describe_tasks` during watch but it seems brittle.

File: prefect_aws/ecs.py [hunk 1]
```diff
@@ -701,9 +701,17 @@ def _watch_task_run(
...
+            tasks = ecs_client.describe_tasks(tasks=[task_arn], cluster=cluster_arn)[
                 "tasks"
-            ][0]
+            ]
+
+            if not tasks:
+                # Intermittently, the task will not be described
+                self.logger.debug(f"{self._log_prefix}: Task not found.")
+                time.sleep(self.task_watch_poll_interval)
+                continue
```
Comment (desertaxle) [inline prefect_aws/ecs.py:712]: Could we include a timeout check here to avoid running too long if the task is never found for some reason?

Comment (madkinsz) [reply prefect_aws/ecs.py:712]: Hm true we do bypass the timeout check, I'll move this around.

**Review by desertaxle — APPROVED** [review]
LGTM! I think we'll be fine without a test for this; the effort doesn't appear to be worth the benefit

**Pull request merged**